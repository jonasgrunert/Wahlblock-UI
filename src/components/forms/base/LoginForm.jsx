import * as React from 'react';
import PropTypes from 'prop-types';
import { Field as ReduxField } from 'redux-form';
import { Button, Columns, Column, Control, Field, Input as BInput, Label } from 'bloomer';

export const TextInput = field => <BInput {...field.input} />;

TextInput.propTypes = {
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  state: PropTypes.bool,
};

TextInput.defaultProps = {
  state: false,
};

export class LoginForm extends React.Component {
  componentWillMount() {
    this.props.generateKey();
  }

  render() {
    return (
      <form onSubmit={this.props.handleSubmit}>
        <Field>
          <Label>Name</Label>
          <Control>
            <ReduxField
              name="name"
              component={TextInput}
              type="text"
              placeholder="Max Mustermann"
            />
          </Control>
        </Field>
        <Field>
          <Label>PIN</Label>
          <Control>
            <ReduxField
              name="pin"
              component={TextInput}
              type="password"
              placeholder="Your PIN"
              state={this.props.state}
            />
          </Control>
        </Field>
        <Columns isMobile>
          <Column isSize="full">
            <Button
              type="submit"
              isColor="primary"
              isSize="medium"
              isPulled="right"
              isOutlined
              isLoading={this.props.state}
            >
              Login
            </Button>
          </Column>
        </Columns>
      </form>
    );
  }
}

LoginForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  generateKey: PropTypes.func.isRequired,
  state: PropTypes.bool,
};

LoginForm.defaultProps = {
  state: false,
};
