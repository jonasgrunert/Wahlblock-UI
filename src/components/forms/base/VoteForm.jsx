import * as React from 'react';
import PropTypes from 'prop-types';
import { Field as ReduxField } from 'redux-form';
import { Button, Control, Column, Columns, Field, Label, Radio } from 'bloomer';

const Option = field => <Radio {...field.input}> {field.input.value} </Radio>;

const Options = optionsConfig =>
  optionsConfig.map(config => (
    <span>
      <ReduxField
        name="selection"
        component={Option}
        type="radio"
        option={config.option}
        value={config.option}
      />
      <br />
    </span>
  ));

export const Vote = props => (
  <form onSubmit={props.handleSubmit}>
    <Field>
      <Label hasTextAlign="left">Your Selection</Label>
      <Control>{Options(props.options)}</Control>
    </Field>
    <Columns isMobile>
      <Column isSize="full">
        <Button
          type="submit"
          isColor="primary"
          isSize="medium"
          isPulled="right"
          isLoading={props.state}
        >
          Vote
        </Button>
      </Column>
    </Columns>
  </form>
);

Vote.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  state: PropTypes.bool,
  options: PropTypes.arrayOf(PropTypes.string),
};

Vote.defaultProps = {
  state: false,
  options: ['Ja', 'Nein'],
};
