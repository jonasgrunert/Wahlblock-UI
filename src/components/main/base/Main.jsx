import { Box, Column, Columns, Container, Notification, Title } from 'bloomer';
import * as React from 'react';
import PropTypes from 'prop-types';

import LoginFormContainer from '../../forms/hoc/loginForm';

export class Main extends React.Component {
  componentDidMount() {
    this.props.redirect();
  }

  componentDidUpdate() {
    this.props.redirect();
  }

  render() {
    return (
      <Container hasTextAlign="centered">
        <Columns isCentered>
          <Column isSize="1/2">
            <Box hasTextAlign="left">
              <Title hasTextColor="primary">Login</Title>
              <Notification isColor="danger">
                Once logged in on one device you have to complete the vote process.
              </Notification>
              <LoginFormContainer />
            </Box>
          </Column>
        </Columns>
      </Container>
    );
  }
}

Main.propTypes = {
  redirect: PropTypes.func.isRequired,
};
