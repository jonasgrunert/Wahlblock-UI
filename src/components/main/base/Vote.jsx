import { Box, Container, Column, Columns, Title } from 'bloomer';
import * as React from 'react';
import PropTypes from 'prop-types';

import VoteFormWrapper from '../../forms/hoc/voteForm';

// STateful compoent for redirecting upon login
export class Vote extends React.Component {
  componentWillMount() {
    this.props.redirect();
  }

  componentWillUpdate() {
    this.props.redirect();
  }

  render() {
    return (
      <Container hasTextAlign="centered">
        <Columns isCentered>
          <Column isSize="1/2">
            <Box>
              <Title isSize={2} hasTextColor="black">
                Vote now
              </Title>
              <VoteFormWrapper />
            </Box>
          </Column>
        </Columns>
      </Container>
    );
  }
}

Vote.propTypes = {
  redirect: PropTypes.func.isRequired,
};
