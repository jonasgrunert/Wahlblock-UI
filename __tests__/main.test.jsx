import * as React from 'react';
import { TileTitle, Info } from '../src/components/main/base/Info';
import { Main } from '../src/components/main/base/Main';
import { Outcome, GraphqlContainer, OutcomeContainer } from '../src/components/main/base/Outcome';

const title = 'default';

const informationFetchFulfilled = {
  pending: false,
  rejected: false,
  fulfilled: true,
  value: {
    id: 0,
    title: 'Example',
    description: 'This is  placeholder',
    beginDate: '',
    endDate: '',
    selectionOptions: [
      {
        position: 1,
        option: 'None',
      },
    ],
  },
};

const informationFetchPending = {
  pending: true,
  rejected: false,
  fulfilled: false,
};

const informationFetchRejected = {
  pending: false,
  rejected: true,
  fulfilled: false,
  value: {
    id: 0,
    title: 'Example',
    description: 'This is  placeholder',
    beginDate: '',
    endDate: '',
    selectionOptions: [
      {
        position: 1,
        option: 'None',
      },
    ],
  },
};

describe('render info page', () => {
  it('sould render titletitle', () => {
    const tiletitle = shallow(<TileTitle title={title} />);
    expect(tiletitle).toMatchSnapshot();
  });
  it('should render info pending', () => {
    const info = shallow(<Info dispatchInformationGet={jest.fn()} InformationFetch={informationFetchPending} />);
    expect(info).toMatchSnapshot();
  });
  it('should render info rejected', () => {
    const info = shallow(<Info dispatchInformationGet={jest.fn()} InformationFetch={informationFetchRejected} />);
    expect(info).toMatchSnapshot();
  });
  it('should render info fulfilled', () => {
    const info = shallow(<Info dispatchInformationGet={jest.fn()} InformationFetch={informationFetchFulfilled} />);
    expect(info).toMatchSnapshot();
  });
});

describe('render main page', () => {
  it('should render main', () => {
    const main = shallow(<Main redirect={jest.fn()} />);
    expect(main).toMatchSnapshot();
  });
});

const blockchain = {
  count: [100, 20, 40],
  backgroundColor: ['hsl(348, 100%, 61%)', 'hsl(204, 86%, 53%)', 'hsl(48, 100%, 67%)'],
  possibilities: ['green', 'blue', 'red'],
};

describe('render outcome page', () => {
  it('should render outcome', () => {
    const outcome = shallow(<Outcome />);
    expect(outcome).toMatchSnapshot();
  });
  it('should render graphqlcontainer loading', () => {
    const graphqlcontainer = shallow(<GraphqlContainer loading />);
    expect(graphqlcontainer).toMatchSnapshot();
  });
  it('should render graphqlcontainer error', () => {
    const graphqlcontainer = shallow(<GraphqlContainer error data={blockchain} />);
    expect(graphqlcontainer).toMatchSnapshot();
  });
  it('should render graphqlcontainer loading', () => {
    const graphqlcontainer = shallow(<GraphqlContainer data={blockchain} />);
    expect(graphqlcontainer).toMatchSnapshot();
  });
});
