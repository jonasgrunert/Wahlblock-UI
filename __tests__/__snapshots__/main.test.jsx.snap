// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render info page should render info fulfilled 1`] = `
<withHelpersModifiers(Container)
  hasTextAlign="centered"
>
  <withHelpersModifiers(Tile)
    isAncestor={true}
  >
    <withHelpersModifiers(Tile)
      isParent={true}
      isSize={8}
    >
      <withHelpersModifiers(Tile)
        isChild={true}
        render={[Function]}
      />
    </withHelpersModifiers(Tile)>
    <withHelpersModifiers(Tile)
      isParent={true}
      isSize={4}
      isVertical={true}
    >
      <withHelpersModifiers(Tile)
        isChild={true}
        render={[Function]}
      />
      <withHelpersModifiers(Tile)
        isChild={true}
        render={[Function]}
      />
    </withHelpersModifiers(Tile)>
  </withHelpersModifiers(Tile)>
</withHelpersModifiers(Container)>
`;

exports[`render info page should render info pending 1`] = `
<withHelpersModifiers(Container)
  hasTextAlign="centered"
>
  <Loading
    color="#fff"
    delay={1000}
    height={64}
    type="spinningBubbles"
    width={64}
  />
</withHelpersModifiers(Container)>
`;

exports[`render info page should render info rejected 1`] = `
<withHelpersModifiers(Container)
  hasTextAlign="centered"
>
  <withHelpersModifiers(Icon)
    className="fa fa-exclamation-triangle fa-3x"
    isSize="large"
  />
</withHelpersModifiers(Container)>
`;

exports[`render info page sould render titletitle 1`] = `
<withHelpersModifiers(Title)
  hasTextAlign="left"
  hasTextColor="black"
  isSize={2}
>
  default
</withHelpersModifiers(Title)>
`;

exports[`render main page should render main 1`] = `
<withHelpersModifiers(Container)
  hasTextAlign="centered"
>
  <withHelpersModifiers(Columns)
    isCentered={true}
  >
    <withHelpersModifiers(Column)
      isSize="1/2"
    >
      <withHelpersModifiers(Box)
        hasTextAlign="left"
      >
        <withHelpersModifiers(Title)
          hasTextColor="primary"
        >
          Login
        </withHelpersModifiers(Title)>
        <withHelpersModifiers(Notification)
          isColor="danger"
        >
          Once logged in on one device you have to complete the vote process.
        </withHelpersModifiers(Notification)>
        <Connect(Connect(ReactReduxFetch.connect(Connect(ReduxForm)))) />
      </withHelpersModifiers(Box)>
    </withHelpersModifiers(Column)>
  </withHelpersModifiers(Columns)>
</withHelpersModifiers(Container)>
`;

exports[`render outcome page should render graphqlcontainer error 1`] = `
<Loading
  color="#fff"
  delay={1000}
  height={64}
  type="spinningBubbles"
  width={64}
/>
`;

exports[`render outcome page should render graphqlcontainer loading 1`] = `
<Loading
  color="#fff"
  delay={1000}
  height={64}
  type="spinningBubbles"
  width={64}
/>
`;

exports[`render outcome page should render graphqlcontainer loading 2`] = `
<Loading
  color="#fff"
  delay={1000}
  height={64}
  type="spinningBubbles"
  width={64}
/>
`;

exports[`render outcome page should render outcome 1`] = `
<withHelpersModifiers(Columns)
  isCentered={true}
>
  <withHelpersModifiers(Column)
    isSize="3/4"
  >
    <withHelpersModifiers(Box)>
      <withHelpersModifiers(Title)
        hasTextColor="black"
        isSize={2}
      >
        Outcome
      </withHelpersModifiers(Title)>
      <Pie
        data={
          Object {
            "datasets": Array [
              Object {
                "backgroundColor": Array [
                  "hsl(348, 100%, 61%)",
                  "hsl(204, 86%, 53%)",
                  "hsl(48, 100%, 67%)",
                ],
                "data": Array [
                  10,
                  20,
                  30,
                ],
              },
            ],
            "labels": Array [
              "Red",
              "Yellow",
              "Blue",
            ],
          }
        }
        options={
          Object {
            "legend": Object {
              "position": "right",
            },
          }
        }
      />
    </withHelpersModifiers(Box)>
  </withHelpersModifiers(Column)>
</withHelpersModifiers(Columns)>
`;
